x/gov/module.go

```diff
@@ -135,8 +135,18 @@ func (am AppModule) BeginBlock(ctx sdk.Ctx, req abci.RequestBeginBlock) {
 
 	u := am.keeper.GetUpgrade(ctx)
 
+	verComp, err := sdk.CompareVersionStrings(ctx.AppVersion(), u.Version)
+	if err != nil {
+		ctx.Logger().Error("INVALID VERSION STRING",
+			"ctx", ctx.AppVersion(),
+			"upgrade", u.Version,
+			"err", err.Error(),
+		)
+		os.Exit(1)
```

Comment:
can we panic or do a Fatal log above instead of os.Exit(1)?

Reply:
`ctx.Logger()` is tendermint's logger that does not have `Fatal`.  The combination of `Error` and `os.Exit(1)` is consistent with the below cases when `os.FindProcess` or `p.Signal` fails.